File: codegen/codegen-erlang/src/test/resources/ir/aws_query_form_decode_aws.expected.erl
parse_query_params(Body) ->
    maps:from_list(uri_string:dissect_query(Body)).

form_value(Params, Key) ->
    maps:get(Key, Params, undefined).

form_list_values_aws(Params, Key) ->
    Prefix = <<Key/binary, ".member.">>,
    indexed_form_values(Params, Prefix).

indexed_form_values(Params, Prefix) ->
    Entries = [
        {form_index(K, Prefix), maps:get(K, Params)}
     || K <- maps:keys(Params), binary:match(K, Prefix) =:= {0, byte_size(Prefix)}
    ],
    case lists:sort(Entries) of
        [] -> undefined;
        Sorted -> [V || {_, V} <- Sorted]
    end.

form_index(Key, Prefix) ->
    Rest = binary:part(Key, byte_size(Prefix), byte_size(Key) - byte_size(Prefix)),
    binary_to_integer(Rest).

File: codegen/codegen-elixir/src/test/resources/ir/aws_query_form_decode_aws.expected.ex
defp parse_query_params(body) do
  body
  |> URI.decode_query()
  |> Map.new()
end

defp form_value(params, key), do: Map.get(params, key)

defp form_list_values_aws(params, key) do
  prefix = <<key, ".member.">>
  indexed_form_values(params, prefix)
end

defp indexed_form_values(params, prefix) do
  values =
    params
    |> Enum.filter(fn {k, _} -> String.starts_with?(k, prefix) end)
    |> Enum.sort_by(fn {k, _} -> String.to_integer(String.replace_prefix(k, prefix, "")) end)
    |> Enum.map(fn {_, v} -> v end)
  case values do
    [] -> nil
    values -> values
  end
end
